migrations/steps.py

```diff
@@ -0,0 +1,139 @@
+"""
+General purpose migration steps
+"""
+
+import shutil
+import sys
+import tempfile
+from pathlib import Path
+
+
+class MigrationStep:
+    def validate(self):
+        return True
```

Comment:
I'd suggest `NotImplementedError` or `false` as a default if these validation, snapshot etc methods are a contract that any migration steps must themselves implement

Reply:
The most recent version uses Abstract Base Classes to enforce implementing the required methods - validation, snapshots and rollback are optional.